src/enclave-agent/src/main.rs

```diff
@@ -0,0 +1,152 @@
+use std::env;
+use std::path::Path;
+use std::sync::Arc;
+
+use clap::{crate_authors, crate_version, App, Arg};
+//use log::LevelFilter;
+use log::*;
+use protocols::{image, image_ttrpc};
+use ttrpc::asynchronous::Server;
+use ttrpc::{self, error::get_rpc_status as ttrpc_error};
+
+use async_trait::async_trait;
+use tokio::signal::unix::{signal, SignalKind};
+use tokio::sync::Mutex;
+
+use anyhow::{anyhow, Result};
+use image_rs::image::ImageClient;
+
+const CONTAINER_BASE: &str = "/run/enclave-cc-containers";
+const CC_IMAGE_WORK_DIR: &str = "/run/image/";
+
+// TODO: will replace with unix socket
+const SOCK_ADDR: &str = "tcp://0.0.0.0:7788";
+
+struct ImageService {
+    image_client: Arc<Mutex<ImageClient>>,
+}
+
+// A container ID must match this regex:
+//     ^[a-zA-Z0-9][a-zA-Z0-9_.-]+$
+fn verify_cid(id: &str) -> Result<()> {
```

Comment:
I seem to remember writing this code (and lots of unit tests) for the Kata Containers agent: see https://github.com/kata-containers/kata-containers/pull/1521 :smile:

It might be a good idea to mention this code was taken from the Kata Containers agent and include details of which commit the code was copied from in your commit.

Reply:
Reflecting some verbal discussion with @jodh-intel ... For purposes of linking patches in future issues it might be best to put that reference in the code comments here (rather than the commit message). A longer term fix might be breaking this out as a separate module both projects can depend on.

Reply:
> I seem to remember writing this code (and lots of unit tests) for the Kata Containers agent: see [kata-containers/kata-containers#1521](https://github.com/kata-containers/kata-containers/pull/1521) 😄
> 
> It might be a good idea to mention this code was taken from the Kata Containers agent and include details of which commit the code was copied from in your commit.

Thanks James, we are modifying this issue.

Reply:
> Reflecting some verbal discussion with @jodh-intel ... For purposes of linking patches in future issues it might be best to put that reference in the code comments here (rather than the commit message). A longer term fix might be breaking this out as a separate module both projects can depend on.

got it, thanks dan